### PR TITLE
docs: add doc on configuring custom services

### PR DESCRIPTION
**Pull request by achingbrain**
Writes up the perceived wisdom on service implementation.

## Change checklist

- [ ] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: doc/SERVICES.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    myService: myService()
+  }
+})
+
+console.info(node.services.myService.saySomething()) // 'Hello, my peer id is 123Koo....'
+```
+
+### Using classes
+
+Use a class for your service and returning an instance will allow you to keep your code manageable.
```
Comment (maschad) [inline doc/SERVICES.md:125]: I'm not sure it's appropriate to suggest a particular programming paradigms here, it seems beyond the scope of documenting configuration.

File: doc/SERVICES.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    })
+  }
+})
+
+console.info(node.services.myService.saySomething()) // 'World'
+```
+
+### Accessing libp2p components
+
+Services can access internal libp2p components such as the address manger and connection manager by accepting an argument to the returned function.
```
[the remark this answers is not in the document]
Comment (achingbrain) [reply doc/SERVICES.md]: Sounds good.  I wanted to do something simple but I guess this gives a clue of where to find the internal interface definitions.

File: doc/SERVICES.md (hunk 1 above)
Comment (achingbrain) [reply doc/SERVICES.md]: Most people will just copy/paste so I think it's fair to show them the "most idiomatic" way of doing things, though you are right it's a bit weird to call this out specifically.

I've updated the examples.

**Review by maschad — COMMENTED** [review]
Nice!

**Comment by 2color** [comment]
This is really nice. Just saw this now!❤️

**Pull request merged**